modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java

```diff
@@ -250,6 +252,29 @@ public void testClientInfoIgniteClientFail() throws Exception {
         }
     }
 
+    /**
+     *
+     */
+    @Test
+    public void testClientInfoGridClientNotFail() throws Exception {
+        Ignite ignite = startGrids(2);
+
+        assertEquals(2, ignite.cluster().topologyVersion());
+
+        ignite.cluster().state(ClusterState.ACTIVE);
+
+        try (GridClient client = GridClientFactory.start(getGridClientConfiguration())) {
+            assertTrue(client.connected());
+
+            GridClientClusterState state = client.state();
+
+            // Close a coordinator to force the client to send a CLUSTER_CURRENT_STATE message to the other node.
```

Comment:
... "in the next statement", I'd like to add. It's not so obvious from the first glance, that state.state() causes some communication with cluster.

Reply:
added this to the comment as you suggested